covalent/_results_manager/results_manager.py

```diff
@@ -49,59 +59,114 @@ def get_result(
         result_object: The result from the file.
 
     Raises:
-        RuntimeError: If the result is not ready to read yet.
-        FileNotFoundError: If the result file is not found.
+        MissingLatticeRecordError: If the result file is not found.
     """
 
     try:
-        result_object = _get_result_from_file(dispatch_id, results_dir, wait)
-
-    except FileNotFoundError:
-        raise FileNotFoundError(
-            f"Result was not found in the specified directory. Please make sure it hasn't been moved or try a different directory than {results_dir}."
+        result_object = _get_result_from_db(
+            DataStore(db_URL=f"sqlite+pysqlite:///{_db_path()}"),
+            dispatch_id,
+            wait,
         )
 
+    except MissingLatticeRecordError as e:
+        raise MissingLatticeRecordError(
+            f"Dispatch ID {dispatch_id} was not found in the database. Either the Dispatch ID is incorrect or wait a couple of seconds before trying again."
+        ) from e
+
     return result_object
 
 
-def _get_result_from_file(
-    dispatch_id: str, results_dir: str = get_config("dispatcher.results_dir"), wait: bool = False
+def result_from(lattice_record: Lattice) -> Result:
+    with open(
+        os.path.join(lattice_record.storage_path, lattice_record.function_filename), "rb"
+    ) as f:
+        function = pickle.loads(f.read())
+    with open(
+        os.path.join(lattice_record.storage_path, lattice_record.executor_filename), "rb"
+    ) as f:
+        executor = pickle.loads(f.read())
+    with open(
+        os.path.join(lattice_record.storage_path, lattice_record.inputs_filename), "rb"
+    ) as f:
+        inputs = pickle.loads(f.read())
+    with open(os.path.join(lattice_record.storage_path, lattice_record.error_filename), "rb") as f:
+        error = pickle.loads(f.read())
+    with open(
+        os.path.join(lattice_record.storage_path, lattice_record.transport_graph_filename), "rb"
+    ) as f:
+        transport_graph = pickle.loads(f.read())
+    with open(
+        os.path.join(lattice_record.storage_path, lattice_record.results_filename), "rb"
+    ) as f:
+        output = pickle.loads(f.read())
+
+    attributes = {
+        "metadata": {"executor": executor},
+        "args": inputs["args"],
+        "kwargs": inputs["kwargs"],
```

Comment:
fyi, the lattice object now has two [additional attributes](https://github.com/AgnostiqHQ/covalent/blob/378131b777de2185c70c52a483cdfa54c3952875/covalent/_workflow/lattice.py#L94-L95): `named_args` and `named_kwargs`. These were previously computed by  `result_webhook.py` with each node update from the lattiice args and kwargs, but doing so requires access to the unserialized workflow inputs. Hence they are now [stored](https://github.com/AgnostiqHQ/covalent/commit/f06f84a0cc487a85ac88177c903f898fe317f845#diff-bfbac3709c395f2f264b85d0a49527ee0e1f11a52a1e2484d28f4fc867b82744R231) in the lattice during build_graph and simply [read by `result_webhook.py`](https://github.com/AgnostiqHQ/covalent/commit/f06f84a0cc487a85ac88177c903f898fe317f845#diff-6a95a7b8ccaf3877a43ca2eb8debac479d7cfecd82a12699b82d8239511137c7L86)

Reply:
[This](https://github.com/AgnostiqHQ/covalent/blob/378131b777de2185c70c52a483cdfa54c3952875/covalent/_results_manager/result.py#L728) is the problematic line. `Result.result` tries to deserialize the result and should never be called by the server. Instead use `Result.encoded_result`.

Reply:
Are there any actionable items on my side or can it wait?

Reply:
Might be an issue for the UI which pulls the result object from DB - [`encode_result`](https://github.com/AgnostiqHQ/covalent/issues/748) won't work. But I'd say that if tests pass, this can be deferred to v10 or v11 of the schema -- whenever the UI needs to work.